### PR TITLE
Code cleanups: part 2: implement cleanups Codacy & cppcheck

### PR DESCRIPTION
**Pull request by commodo**
This changeset implements some changes reported in  Codacy tool and cppcheck.

It's part 2 of PR #98 
The intent is to send small PRs to make things easier to review.

Signed-off-by: Alexandru Ardelean <alexandru.ardelean@analog.com>

### CODE REVIEW

**Comment by commodo** [comment]
If no objections, will merge later today

**Comment by lclausen-adi** [comment]
You need to allow more time for review. Don't merge your own patches without having had review from somebody else.

**Comment by commodo** [comment]
Ack

File: phone_home.c [hunk 1]
```diff
@@ -260,7 +261,6 @@ Release * release_get_latest(void)
 
 	/* Get the release SHA commit */
 	json_t *j_tags, *tag, *name, *commit;
-	char *tag_name;
 	size_t i;
 
 	tag_name = strdup(json_string_value(json_object_get(j_release, "tag_name")));
```
Comment (lclausen-adi) [inline phone_home.c]: I'd say it is probably safe to just drop the strdup().

File: xml_utils.c [hunk 2]
```diff
@@ -88,7 +88,7 @@ char **get_xml_list(char * buf_dir_name, int *list_size)
 		return NULL;
 	}
 
-	while (ent = readdir(d), ent != NULL) {
+	while ((ent = readdir(d))) {
```
Comment (lclausen-adi) [inline xml_utils.c]: The extra parenthesis are probably not needed.

File: phone_home.c [hunk 3]
```diff
@@ -263,7 +263,7 @@ Release * release_get_latest(void)
 	char *tag_name;
```
Comment (lclausen-adi) [inline phone_home.c]: Looking at the API documentation for json_string_value() tag_name should also be made const here.

File: xml_utils.c (hunk 2 above)
Comment (commodo) [reply xml_utils.c]: seems it's needed based on default GCC options;
build fails with:
```
/home/travis/build/analogdevicesinc/iio-oscilloscope/xml_utils.c: In function ‘get_xml_list’:
/home/travis/build/analogdevicesinc/iio-oscilloscope/xml_utils.c:91:2: error: suggest parentheses around assignment used as truth value [-Werror=parentheses]
  while (ent = readdir(d)) {
  ^
```

will add it back

**Comment by commodo** [comment]
Changelog v2 -> v3:
* add const qualifier to `tag_name`
* added parentheses back to `ent = readdir()`

File: phone_home.c (hunk 3 above)
Comment (commodo) [reply phone_home.c]: seems that a compiler check also agrees here about the `const`

**Comment by commodo** [comment]
if no other objections will merge this later today [ in ~3-4h ]

File: phone_home.c (hunk 3 above)
Comment (lclausen-adi) [reply phone_home.c]: where and how? as far as I can see tag_name is only used for strcmp()

Comment (commodo) [reply phone_home.c]: This build: https://travis-ci.org/commodo/iio-oscilloscope/jobs/420955832
Apparently I stopped it before it finished ; I changed the `ent = readdir()` parentheses, then added the `const` qualifier (based on your comment), then stopped that build (to let a new one run).

**Comment by lclausen-adi** [comment]
Either wait until there are two approvals or two days have passed and nobody complained.

File: phone_home.c (hunk 3 above)
Comment (lclausen-adi) [reply phone_home.c]: ok, sorry, misunderstood. Thought you meant the compiler did not agree.

**Comment by commodo** [comment]
ack

**Comment by commodo** [comment]
Will merge this today if no other objections

**Pull request merged**